### PR TITLE
Handling presto Unnest cross join to Databricks lateral view

### PR DESCRIPTION
**Pull request by ganeshdogiparthi-db**

### CODE REVIEW

File: src/databricks/labs/remorph/snow/databricks.py [hunk 1]
```diff
@@ -87,7 +87,46 @@ def _select_contains_index(expression: exp.Select) -> bool:
...
+def _generate_function_str(select_contains_index, has_parse_json, generator_expr, alias, is_outer, alias_str):
+    if select_contains_index:
+        generator_function_str = f"POSEXPLODE({generator_expr})"
+        alias_str = f"{' ' + alias.name if isinstance(alias, exp.TableAlias) else ''} AS index, value"
+    elif has_parse_json and is_outer:
+        generator_function_str = f"VARIANT_EXPLODE_OUTER({generator_expr})"
+    elif has_parse_json:
+        generator_function_str = f"VARIANT_EXPLODE({generator_expr})"
+    else:
+        generator_function_str = f"VIEW EXPLODE({generator_expr})"
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/snow/databricks.py:112]: @ganeshdogiparthi-db can you add couple of functional tests for Snowflake to cover view explode

Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/snow/databricks.py:112]: Added Sundar 
Screenshot 2024-11-18 at 1 08 28 PM
Screenshot 2024-11-18 at 1 07 59 PM
Screenshot 2024-11-18 at 1 07 44 PM
Screenshot 2024-11-18 at 1 07 38 PM
Screenshot 2024-11-18 at 12 24 59 PM
Screenshot 2024-11-18 at 12 24 53 PM

**Review by sundarshankar89 — APPROVED** [review]
LGTM

File: core/src/test/scala/com/databricks/labs/remorph/coverage/AcceptanceSpec.scala [hunk 2]
```diff
@@ -34,7 +34,11 @@ class SnowflakeAcceptanceSuite
             "test_command/test_command_3.sql",
             "test_skip_unsupported_operations/test_skip_unsupported_operations_7.sql",
             "test_skip_unsupported_operations/test_skip_unsupported_operations_9.sql",
-            "test_skip_unsupported_operations/test_skip_unsupported_operations_10.sql"),
+            "test_skip_unsupported_operations/test_skip_unsupported_operations_10.sql",
+            // TODO - Fix these tests as part of the lateral view
+            "arrays/test_array_construct_1.sql",
```
Comment (sundarshankar89) [inline core/src/test/scala/com/databricks/labs/remorph/coverage/AcceptanceSpec.scala:39]: @gueniai FYI.. this is an exception to the usual process, we will take the ANTLR enhancements for this in separate PR.

**Review by bishwajit-db — APPROVED** [review]
LGTM

**Pull request merged**